heron/schedulers/src/java/com/twitter/heron/scheduler/yarn/HeronMasterDriver.java

```diff
@@ -464,17 +503,41 @@ private void launchScheduler() {
   class ContainerAllocationHandler implements EventHandler<AllocatedEvaluator> {
     @Override
     public void onNext(AllocatedEvaluator evaluator) {
-      LOG.log(Level.INFO, "New container received, id: {0}", evaluator.getId());
-      try {
-        allocatedContainerQ.put(evaluator);
-        LOG.log(Level.INFO, "{0} containers waiting for activation", allocatedContainerQ.size());
-      } catch (InterruptedException e) {
-        // Fail deployment of topology if there is a error starting any worker
-        LOG.log(Level.SEVERE, "Container allocation for id:{0} failed. Trying to close currently"
-            + " allocated resources for {1}", new Object[]{evaluator.getId(), topologyName});
-        killTopology();
-        throw new RuntimeException("Failed to allocate container for workers", e);
+      EvaluatorDescriptor descriptor = evaluator.getEvaluatorDescriptor();
+      LOG.log(Level.INFO, String.format("New container received, id: %s, mem: %d, cores: %d",
+          evaluator.getId(), descriptor.getMemory(), descriptor.getNumberOfCores()));
+
+      Optional<HeronWorker> result;
```

Comment:
Yes, it could be inside. Just to be sure, did you observe any functional issues with it other than code readability?